### PR TITLE
Add Ctrl-D to Text Editor

### PR DESCRIPTION
**Pull request by Harm10**

### CODE REVIEW

File: src/robotide/editor/texteditor.py [hunk 1]
```diff
@@ -85,6 +85,7 @@ def f(event):
         self.register_shortcut('CtrlCmd-Z', focused(lambda e: self._editor.undo()))
         self.register_shortcut('CtrlCmd-Y', focused(lambda e: self._editor.redo()))
         # self.register_shortcut('Del', focused(lambda e: self._editor.delete()))
+        self.register_shortcut('CtrlCmd-D', focused(lambda e: self._editor.delete_row(e)))
```
Comment (HelioGuilherme66) [inline src/robotide/editor/texteditor.py:88]: Using this registering does not overrides the Ctrl-D from Scintilla library, and the command is "duplicate text". My PR has this line commented.

File: src/robotide/editor/texteditor.py [hunk 2]
```diff
@@ -973,6 +974,26 @@ def _enclose_text(open_symbol, value=''):
...
 
+    def delete_row(self, event):
+        start, end = self._editor.GetSelection()
+        cursor = self._editor.GetCurrentPos()
+        ini_line = self._editor.LineFromPosition(start)
+        end_line = self._editor.LineFromPosition(end)
+        begpos = self._editor.PositionFromLine(ini_line)
+        endpos = self._editor.PositionFromLine(end_line+1)
+        self._editor.SelectNone()
+        self._editor.Remove(begpos, endpos)
```
Comment (HelioGuilherme66) [inline src/robotide/editor/texteditor.py:985]: In my PR I changed this Remove, to LineDelete.

**Review by HelioGuilherme66 — APPROVED** [review]
This will be merged as soon you merge my PR in your fork.
Thanks.

File: src/robotide/editor/texteditor.py (hunk 2 above)
Comment (Harm10) [reply src/robotide/editor/texteditor.py]: What is the advantage of LineDelete over Remove?

**Pull request merged**